src/App.js

```diff
@@ -90,21 +99,58 @@ class Root extends React.Component {
             }
           }
         }
-        this.props.historyUpdate({'data': [wsData.history]});
+        wallet.newAddressHistory(wsData.history);
       } else {
         console.log('Websocket message not handled. Type:', wsData.type);
       }
     }
   }
 
   handleWebsocketStorage = (wsData) => {
-    if (wallet.loaded()) {
+    if (hathorLib.wallet.loaded()) {
       // We are still receiving lot of ws messages that are destined to the admin-frontend and not this wallet
       // TODO separate those messages
       console.log('Websocket message not handled. Type:', wsData.type);
     }
   }
 
+  /**
+   * Method called when WebSocket receives a message after loading address history
+   * We just check and save the version that was loaded and update redux data
+   *
+   * @param {Object} data Object with {'historyTransactions', 'addressesFound'}
+   */
+  addressesLoadedUpdate = (data) => {
+    // Update the version of the wallet that the data was loaded
+    localStorage.setItem('wallet:version', VERSION);
+
+    // Check api version everytime we load address history
+    version.checkApiVersion();
+
+    // Update redux with loaded data
+    store.dispatch(dataLoaded({ addressesFound: data.addressesFound, transactionsFound: Object.keys(data.historyTransactions).length }));
+  }
+
+  /**
+   * Method called when WebSocket updates isOnline attribute, so we update this parameter in redux
+   *
+   * @param {boolean} data Boolean if websocket is online
+   */
+  isOnlineUpdate = (data) => {
+    store.dispatch(isOnlineUpdate({ isOnline: data }));
+  }
+
+  /**
+   * Method called when need to reload data when websocket reconnects
+   */
+  reloadData = (data) => {
+    // Error modal might be opened, then we need to close it when reloading data
+    // Sometimes when the error modal is openend, it leaves only the backdrop after the reload
+    // That's why we need to close it manually
+    $('#requestErrorModal').modal('hide');
```

Comment:
It is part of another PR, right? Maybe we should remove it from this one.

Reply:
It was in a different file in the other PR but now I have done a different solution, so this code here was removed.

Reply:
Ok!